js/src/common/utils/mapRoutes.ts

```diff
@@ -7,7 +7,7 @@
  * @param {String} [basePath]
  * @return {Object}
  */
-export default function mapRoutes(routes, basePath = '') {
```

Comment:
We'll want to define a custom interface for the routes input. For the output, Mithril might have something, but if not we'll need something custom there too.

Reply:
Using `mithril.Route` as type for routes and `Partial<mithril.Route>` as return type.

Reply:
The input isn't mithril.Route: mithril's Route maps route to component; our route abstraction maps route names to an object containing route paths and the component (and possibly route resolvers in the future, but under their own keys). A custom interface would be preferable for that input.

Also, I believe Mithril uses mithril.RouteDefs, not mithril.Route, for the actual route definitions